src/mon/OSDMonitor.cc

```diff
@@ -519,6 +519,10 @@ int OSDMonitor::reweight_by_utilization(int oload,
 	if (*q >= (int)pgs_by_osd.size())
 	  pgs_by_osd.resize(*q);
 	if (pgs_by_osd[*q] == 0) {
+          if (osdmap.crush->get_item_weightf(*q) <= 0) {
```

Comment:
And I doubt we shall let this happen, otherwise we will encounter an **division by zero** error at https://github.com/ceph/ceph/pull/8941/files#diff-0a5db46a44ae9900e226289a810f10e8R611